src/connectedk8s/azext_connectedk8s/_troubleshootutils.py

```diff
@@ -0,0 +1,990 @@
+            deployments_path = os.path.join(filepath_with_timestamp, "Arc_Deployment_Logs")
+            try:
+                os.mkdir(deployments_path)
+            except FileExistsError:
+                pass
+
+            # To retrieve all the the deployment that are present in the Cluster
+            deployments_list = appv1_api_instance.list_namespaced_deployment("azure-arc")
+
+            # Traversing through all the deployments present
+            for deployment in deployments_list.items:
+
+                # Fetching the deployment name
+                deployment_name = deployment.metadata.name
+
+                arc_deployment_logs_path = os.path.join(deployments_path, deployment_name + ".txt")
+                # Creating a text file with the name of the deployment and adding deployment status in it
+                with open(arc_deployment_logs_path, 'w+') as deployment_file:
+                    deployment_file.write(str(deployment.status))
+
+        return consts.Diagnostic_Check_Passed, storage_space_available
+
+    # For handling storage or OS exception that may occur during the execution
+    except OSError as e:
+        if "[Errno 28]" in str(e):
+            storage_space_available = False
+            telemetry.set_exception(exception=e, fault_type=consts.No_Storage_Space_Available_Fault_Type, summary="No space left on device")
+            shutil.rmtree(filepath_with_timestamp, ignore_errors=False, onerror=None)
+        else:
+            logger.warning("An exception has occured while trying to fetch the azure arc deployment logs from the cluster. Exception: {}".format(str(e)) + "\n")
+            telemetry.set_exception(exception=e, fault_type=consts.Arc_Deployments_Logger_Fault_Type, summary="Error occured in deployments logger")
+            diagnoser_output.append("An exception has occured while trying to fetch the azure arc deployment logs from the cluster. Exception: {}".format(str(e)) + "\n")
+
+    # To handle any exception that may occur during the execution
+    except Exception as e:
+        logger.warning("An exception has occured while trying to fetch the azure arc deployment logs from the cluster. Exception: {}".format(str(e)) + "\n")
+        telemetry.set_exception(exception=e, fault_type=consts.Arc_Deployments_Logger_Fault_Type, summary="Error occured in deployments logger")
+        diagnoser_output.append("An exception has occured while trying to fetch the azure arc deployment logs from the cluster. Exception: {}".format(str(e)) + "\n")
+
+    return consts.Diagnostic_Check_Failed, storage_space_available
+
+
+def check_agent_state(corev1_api_instance, filepath_with_timestamp, storage_space_available):
+
+    global diagnoser_output
+    all_agents_stuck = True
+    sufficient_resource_for_agents = True
+
+    try:
+
+        # To check if agents are stuck because of insufficient resource
+        sufficient_resource_for_agents = True
```

Comment:
let's add a prefix to this variable as "probable" . Something like probable_sufficient_resource_for_Agents. Reason: we're not 100% sure about resource availability since we're not checking that directly. We're depending on the fact that all agents are pending and container status is none. So, it's a probable case and not a clear cut check